### ISSUE TITLE
[Bug] iOS - Editors do not size correctly with other content inline

### ISSUE THREAD

**Issue by matthewskillington** [xamarin/Xamarin.Forms#9870]
### Description

When an editor has to share the screen width with other content either side of it, for example  other grid columns it doesn't seem to size itself correctly vertically to display all of the content within it

### Steps to Reproduce

1. Use the reproduction link to fire up the app onto an iOS device
2. Start the app and start typing into the editor

### Expected Behavior
After 3 or 4 lines you will notice the editor doesn't grow vertically enough to display its content

### Actual Behavior
The editor should grow to show all of the content within it

### Basic Information

- Version with issue: 4.3.0991211
- Last known good version: ?
- IDE: VS2019
- Platform Target Frameworks: 
  - iOS:  13.X
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

![image](https://user-images.githubusercontent.com/14994240/76087784-c149aa00-5fae-11ea-9000-caac5d8e0bcc.png)
![image](https://user-images.githubusercontent.com/14994240/76088764-7e88d180-5fb0-11ea-9048-fad302a5c9e1.png)

### Reproduction Link

[AutoSizeBugGrid.zip](https://github.com/xamarin/Xamarin.Forms/files/4298548/AutoSizeBugGrid.zip)

### Workaround

Using a custom control and overriding the OnMeasure and adjusting the width to account for any content either side of the editor

**Comment by hartez** [comment]
Yeah, the resizing _tries_ to happen on both platforms, but it's all over the place. Sometimes the Editor resizes as you type more characters, sometimes it resizes to add _part_ of a new line when you hit Enter.
